hs_core/hydroshare/resource.py

```diff
@@ -369,31 +369,30 @@ def create_resource(
 
         add_resource_files(resource.short_id, *files)
 
-        if 'owner' in kwargs:
-            owner = utils.user_from_id(kwargs['owner'])
-            resource.owners.add(owner)
-
-        resource.owners.add(owner)
+        resource_access = ResourceAccess(resource=resource)
+        resource_access.save()
+        UserResourcePrivilege(resource=resource_access, grantor=owner.uaccess, user=owner.uaccess,
+                              privilege=PrivilegeCodes.OWNER).save()
 
         if edit_users:
             for user in edit_users:
                 user = utils.user_from_id(user)
-                resource.edit_users.add(user)
```

Comment:
I am even not sure how we can run two different access control logic side by side. I would prefer to move everything ( I think I have mostly done that already) to the new system and just have one access control system in place.